File: api/src/main/java/com/alten/remotesync/application/project/record/response/ProjectDTO.java
package com.alten.remotesync.application.project.record.response;

import com.alten.remotesync.domain.client.model.Client;
import com.alten.remotesync.domain.project.enumeration.ProjectStatus;

import java.time.LocalDate;
import java.util.Date;
import java.util.UUID;

public record ProjectDTO(


        UUID projectId,

        String label,

        String titre,

        ProjectStatus status,

        LocalDate deadLine,
        LocalDate startDate,

        boolean isDeleted,

        Client owner
) {
}
    

File: app/src/app/dto/project.dto.ts
import { ProjectStatus } from '../dto/project-status.enum';
import { Client } from './client.dto';

export interface ProjectDTO {
  projectId: string;
  label: string;
  status: ProjectStatus;
  deadLine: Date;
  startDate: Date;
  isDeleted: boolean;
  owner: Client;
}
